### DISCUSSION TITLE
ExplorerPatcher Broke Windows on KB5023706 Update

### DISCUSSION THREAD

**Discussion by MACinBell** [valinet/ExplorerPatcher#1101] [General]
This broke my W11 installation when I updated this week.  Took a long time to figure out the problem.  W11 booted, but the UI and desktop was a mess and was not functional.  Only thing that partially worked was Task Manager.  Solved problem after going to an earlier restore point, uninstalling ExplorerPatcher, and then installing the update.  Explorerpatcher appears to alter basic W11 files rather than just run in the background.  Because of this, I could find no way to kill the program in Task Manager in order to allow W11 to fully boot.  As much as I like this little program, I don't think I will reinstall as the Dev recommends.  It is just too risky.  Open Shell is still working.

**Comment by dd86k** [comment]
Yep, booted Windows 11 this morning to a black screen after logging in. Windows had installed updates prior to this (including [KB5023706](https://support.microsoft.com/en-us/topic/march-14-2023-kb5023706-os-build-22621-1413-9d3f2de5-08e7-4462-8fba-d944201f4ae1))

I was able to start Firefox, start a cmd session with Administrator privileges, and run `ep_setup /uninstall`. Then, File Explorer booted without issue.

It seems like Microsoft touched ExperienceHost and/or removed the Windows 10 theme fallback. That's just my gut feeling.

**Reply by MACinBell** [reply]
I really like EP, but I am wondering how often this kind of problem will occur with future W11 feature updates.  It seems that EP alters or replaces W11 OS files rather than runs as a separate program in the background, which would be easier to kill with TM.  I see that MS has warned users about not using UI modifiers, so I am also concerned about using OpenShell, which I also have.  OpenShell seems to be working okay for now.  OpenShell is a separate program that runs after W11 boots up.

**Comment by Kirimoar** [comment]
I faced the same problem yesterday - as much as I love the ExplorerPatcher, this was REALLY a huge problem. Took me more than an hour to finally fix it (after identifying what even the problem was).

Solution was a rollback to the windows-update before.

However - I was not running the latest EP version when the bug happened.
Does anybody know if this still happens if I'm going to update windows again?
And what could be a solution?

**Reply by valinet** [reply]
I just updated the program to work on 22621.1413 and with a mechanism to detect repeated crashes and break the loop.

**Reply by MACinBell** [reply]
Yes, it was definitely in a big do-loop - continuously alternating between a flashing white screen and a partially populated desktop. Booting in safe mode did not help at all.   My question is what happens when the loop is broken.  Is EP disabled or uninstalled.  Will it start again on the next boot or the next W11 update.  Not sure what kind of program EP is.  Does it change OS files or does it run as a start-up executable or DLL.  I could not find anyway to kill it in Task Manager or eliminate it from auto-runs.

**Reply by topia** [reply]
https://github.com/valinet/ExplorerPatcher/commit/d7e5b7d3c965803d4d52aefd7fedefe11e3518c4#diff-4aae5a03ea2152d405658dbd5f5f12e81e44d5816d6ca737e4a0ab511d8d5d0cR12447-R12460
the current implementation shows a message box with advice, including update, uninstall, and so on.

**Reply by topia** [reply]
Or do you want to know "how to break the loop"? The explorer should start without any ExplorerPatcher patches (with the above message box).

**Reply by valinet** [reply]
Yeah, exactly, I display a message box early on when ExplorerPatcher loads and do not load anything else further in `explorer`, which should stop the crashing. You can take the oportunity to apply a fix or uninstall it for the time being. The point is, it won't loop more than 3 times when crashing now. Every 3rd time, it disables itself until the next restart and notifies you, which is much better than what was in there previously. EP doesn't modify any system files. I probably should do an in-depth explainer on how it works - it just places its files in various places in the system to have it loaded in processes like `explorer`, the Start menu, the other shell components host process in order to patch in various stuff. But no system file is replaced.

**Reply by barwick11** [reply]
Thanks for the update.  It seems to be more stable now, though now I'm having trouble with the start menu not popping up, and alt-tab functionality doesn't work (seems to crash EP).

**Reply by otismv9** [reply]
@barwick11 Have you figured out a solution? Encountering the same issues on my end

**Reply by austinjklim** [reply]
After updating to latest windows update and EP, it seems that the search bar window box position is "fixed" to the default bottom left position, instead of top left where i have my taskbar at. not a big deal but just wanted to let someone know.

**Reply by barwick11** [reply]
> @barwick11 Have you figured out a solution? Encountering the same issues on my end

Nope, haven't found a solution except uninstalling EP for the moment.  Which is infuriating because I hate Windows 11's taskbar so much.
